### PR TITLE
Test up to sage 9.6

### PR DESCRIPTION
**Pull request by rparini**
Sage no longer distributes linux binaries so need a different approach, using conda seems like the best available option.

### CODE REVIEW

**Comment by rparini** [comment]
I'm baffled as to why sage 9.2 takes 2hr to install with Conda so I'll keep using the binary distribution for that version

**Comment by rparini** [comment]
Doesn't help so I'll just use the binaries for 9.2

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -38,3 +33,27 @@ jobs:
...
+        sage-version:
+          - '9.3'
+          - '9.4'
+          - '9.5'
+          - '9.6'
+    steps:
+      - name: Install sage
+        run: |
+          conda config --add channels conda-forge
+          conda install sage=${{ matrix.sage-version }} -y
```
Comment (isuruf) [inline .github/workflows/test.yml:50]: You can use mamba for faster solves

          conda config --add channels conda-forge
          conda install mamba -y
          mamba install sage=${{ matrix.sage-version }} -y

Comment (rparini) [reply .github/workflows/test.yml:50]: Thanks very much for the tip, I would not have guessed that would make such a big difference! #204

**Pull request merged**